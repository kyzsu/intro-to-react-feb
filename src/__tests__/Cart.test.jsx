import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Cart";

test("snapshot dengan keranjang kosong", () => {
  const { asFragment } = render(<Cart cart={[]} />);
  expect(asFragment()).toMatchSnapshot();
});

test("snapshot dengan keranjang ada isinya", () => {
  const { asFragment } = render(
    <Cart
      cart={[
        {
          pizza: {
            id: "hawaiian",
            name: "The Hawaiian Pizza",
            category: "Classic",
            description: "Sliced Ham, Pineapple, Mozzarella Cheese",
            image: "/public/pizzas/hawaiian.webp",
            sizes: {
              S: 10.5,
              M: 13.25,
              L: 16.5,
            },
            size: "L",
            price: 16.5,
          },
        },
        {
          pizza: {
            id: "the_greek",
            name: "The Greek Pizza",
            category: "Classic",
            description:
              "Kalamata Olives, Feta Cheese, Tomatoes, Garlic, Beef Chuck Roast, Red Onions",
            image: "/public/pizzas/the_greek.webp",
            sizes: {
              S: 12,
              M: 16,
              L: 20.5,
            },
            size: "M",
            price: 16,
          },
        },
      ]}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
