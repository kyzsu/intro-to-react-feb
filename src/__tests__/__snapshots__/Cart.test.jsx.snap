// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot dengan keranjang ada isinya 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul />
    <p>
      Total Harga: $NaN
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot dengan keranjang kosong 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul />
    <p>
      Total Harga: $0.00
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;
